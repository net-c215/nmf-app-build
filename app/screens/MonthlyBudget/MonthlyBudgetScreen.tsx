import React, { useState } from "react";
import { View, Slider, ScrollView } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import * as WebBrowser from "expo-web-browser";

import { Text, Button } from "../../components";
import colors from "../../style/colors";
import styles from "./MonthlyBudgetScreen.styles";
import navigationOptions from "./MonthlyBudgetScreen.navigationOptions";
import { t } from "../../utils";

const DEFAULT_MONTHLY_CARBON_BUDGET = 500;
const MIN_MONTHLY_CARBON_BUDGET = 0;
const MAX_MONTHLY_CARBON_BUDGET = 1000;

const MonthlyBudgetScreen = ({ navigation }) => {
  const [value, setValue] = useState(DEFAULT_MONTHLY_CARBON_BUDGET);
  const onPressInfo = () =>
    WebBrowser.openBrowserAsync(
      "https://en.wikipedia.org/wiki/List_of_countries_by_carbon_dioxide_emissions_per_capita"
    );
  const onPressSaveBudget = () => navigation.goBack();

  return (
    <View style={styles.container}>
      <ScrollView>
        <View style={styles.personnalBudgetContainer}>
          <Text.Primary bold>
            {t("MONTHLY_BUDGET_MY_MONTHLY_CARBON_BUDGET")}
          </Text.Primary>
          <Text.Primary lightGray>{Math.round(value) + " kg"}</Text.Primary>
        </View>
        <Slider
          minimumTrackTintColor={colors.linkGreen}
          maximumTrackTintColor={colors.gray}
          thumbTintColor={colors.linkGreen}
          style={styles.slider}
          maximumValue={MAX_MONTHLY_CARBON_BUDGET}
          minimumValue={MIN_MONTHLY_CARBON_BUDGET}
          value={value}
          onSlidingComplete={setValue}
        />
        <View style={styles.worldBudgetContainer}>
          <View style={styles.worldExampleTitle}>
            <Text.Primary bold>
              {t("MONTHLY_BUDGET_CARBON_EMISSIONS_WORLD")}
              <Ionicons
                name="md-information-circle"
                size={26}
                color={colors.linkGreen}
                onPress={onPressInfo}
              />
            </Text.Primary>
          </View>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_LUXEMBOURG")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_UNITED_STATES")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_JAPAN")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_SWEDEN")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_FRANCE")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_CHINA")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_BRAZIL")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_INDIA")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_ETHIOPIA")}
          </Text.Secondary>
        </View>
      </ScrollView>
      <View style={styles.buttonContainer}>
        <Button.Primary onPress={onPressSaveBudget} textType={"Primary"}>
          <Text.Primary white center bold>
            {t("MONTHLY_BUDGET_SAVE")}
          </Text.Primary>
        </Button.Primary>
      </View>
    </View>
  );
};

MonthlyBudgetScreen.navigationOptions = navigationOptions;

export default MonthlyBudgetScreen;
