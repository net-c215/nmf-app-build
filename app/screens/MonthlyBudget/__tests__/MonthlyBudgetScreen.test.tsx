import React from "react";
import renderer from "react-test-renderer";
import MonthlyBudgetScreen from "../MonthlyBudgetScreen";

it("MonthlyBudgetScreen renders correctly", () => {
  const tree = renderer.create(<MonthlyBudgetScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
