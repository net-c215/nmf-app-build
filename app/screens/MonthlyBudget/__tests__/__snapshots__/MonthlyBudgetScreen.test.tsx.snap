// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyBudgetScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "paddingTop": 26,
          }
        }
      >
        <Text.Primary
          bold={true}
        >
          My carbon monthly budget : 
        </Text.Primary>
        <Text.Primary
          lightGray={true}
        >
          500 kg
        </Text.Primary>
      </View>
      <RCTSlider
        disabled={false}
        enabled={true}
        maximumTrackTintColor="#CED4DA"
        maximumValue={1000}
        minimumTrackTintColor="#4BB377"
        minimumValue={0}
        onChange={null}
        onResponderTerminationRequest={[Function]}
        onSlidingComplete={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={null}
        step={0}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
        thumbTintColor="#4BB377"
        value={500}
      />
      <View
        style={
          Object {
            "flex": 1,
            "marginVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginBottom": 8,
            }
          }
        >
          <Text.Primary
            bold={true}
          >
            CO2 emissions per capita per month 
            <Ionicons
              color="#4BB377"
              name="md-information-circle"
              onPress={[Function]}
              size={26}
            />
          </Text.Primary>
        </View>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          Luxembourg : 3.5 tons
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          United States : 1,5 tons
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          Japan : 900 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          Sweden : 595 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          France : 575 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          China : 522 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          Brazil : 208 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          India : 139 kg
        </Text.Secondary>
        <Text.Secondary
          style={
            Object {
              "marginBottom": 6,
            }
          }
        >
          Ethiopia : 8.3 kg
        </Text.Secondary>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 20,
        "left": 14,
        "position": "absolute",
        "right": 14,
      }
    }
  >
    <Button.Primary
      onPress={[Function]}
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        Save
      </Text.Primary>
    </Button.Primary>
  </View>
</View>
`;
